src/dataspaces.jl

```diff
@@ -15,6 +16,18 @@ The following functions have methods defined for `Dataspace` objects
 """
 Dataspace # defined in types.jl
 
+"""
+    HDF5.UNLIMITED
+
+A sentinel value which indicates an unlimited dimension in a
+[`Dataspace`](@ref).
+
+Can be used as an entry in the `max_dims` argument in the [`Dataspace`](@ref)
+constructor or [`create_dataset`](@ref), or as a `count` argument in
+[`BlockRange`](@ref) when selecting virtual dataset mappings.
+"""
+const UNLIMITED = -1
```

Comment:
Could we also interpret a symbol, `:unlimited`, for this purpose?

Reply:
We could. The question is what should `size` return? Defining `HDF5.UNLIMITED = -1` seemed like the easiest option to get backwards compatibility and type stability.